### ISSUE TITLE
Feature request: display bikesharing station information by leveraging gbfs

### ISSUE THREAD

**Issue by MayeulC** [osmandapp/OsmAnd#16448]
# 🚀 feature request

### Description
Bikesharing stations are already present in the OSM dataset, and displayed in OsmAnd.

My feature request is for station status (number of free bike slots and available bikes, closed station, etc) to be displayed in the app by querying that information from external services.

Additionally, one could imagine making use of that data for routing, at some point.

### Describe the solution you'd like

Have a look at https://github.com/MobilityData/gbfs
> The General Bikeshare Feed Specification, known as GBFS, is the open data standard for shared mobility. GBFS makes real-time data feeds in a uniform format publicly available online, with an emphasis on findability. GBFS is intended to make information publicly available online; therefore information that is personally identifiable is not currently and will not become part of the core specification.

This is an open specification for an exchange format for data about those bike-sharing stations.

The above repository also contains [a csv file](https://github.com/MobilityData/gbfs/blob/master/systems.csv) with a list of providers (over 600) and the associated URLs.

### Describe alternatives you've considered

As I am a subscriber of one of these services, I added a quick search that brings up the appropriate POIs for that service. I also sometimes use Openbikesharing, which had its map broken, and as I write this I discover its successor, [BikeSharingHub](https://f-droid.org/en/packages/fr.fdesousa.bikesharinghub/). Its source code is there: https://github.com/francoisfds/BikeSharingHub

I am mostly asking to integrate this inside OsmAnd, but a link to the external app could be interesting too.

Alternatively, I consider using my own bike, which I actually now do most of the time ;)

**Comment by ceever** [comment]
Oppose.

I don't understand how much more OsmAnd is supposed to cover?!? It is already doing a great job! More and more features just make maintaining the app harder and harder. It also makes using the harder and harder. Why compete with apps that are already doing a great job?

For Bike-sharing you will need an app by the sharing provider anyway to rent the bicycle. These apps usually come with the necessary maps. I don't understand why this should be duplicated in OsmAnd. Yeah, it is great that there is an open standard, but something is not automatically meaningful just because it is open.

**Comment by MayeulC** [comment]
> More and more features just make maintaining the app harder and harder

Not necessarily, **this could also be a plugin**. For now, bike sharing stations are displayed as bike rentals, and can be tapped on. Using a special icon, and displaying station usage on pop-up would be nice. Adding filters to POI search would also be interesting.

Usually, there is only one such service per city, so the local operator could be highlighted differently.

I feel like this feature request, while not useful to everybody, is fairly small in scope and would fit fairly well in a plugin, like sea navigation does. Existing APIs are probably enough, except perhaps for the routing part, but it's just a matter of identifying stations with available bikes/spots as departure and arrival points (and updating them).

The nice thing here is that the open API makes it easy to query. Ideally, I'd love for OSMAnd to make a bit more use of online resources (traffic information, closed roads/tunnels, free parking spots, fares, etc.)

> For Bike-sharing you will need an app by the sharing provider anyway to rent the bicycle

I don't use such an app. The service I'm using can be unlocked using a dedicated transports card, or a username/password.
Not every app includes functionalities like routing, anyway.

> I don't understand how much more OsmAnd is supposed to cover?!? It is already doing a great job!

I agree it's doing a good job (somewhat; it's extremely slow to render on my S4, but that's off-topic). Could it be improved still? Certainly. How much is it supposed to cover? I have no idea, but it's free software, so I can make the changes I'd like to see (and upstream is free to accept them).

**Comment by ceever** [comment]
"... it's extremely slow to render on my S4, but that's off-topic" ... now help me guessing why this could be–trust me, it's far from off-topic! 🤔😅

**Comment by mikehgentry** [comment]
If it were just for one company or region, I could see the objection. If it's in widespread use, it seems like a good idea to me since it's an open standard.

The major issue with 'just get the app' is you can be subscribed to / a user of several of these services (especially if they're pay as you go), so that quickly becomes unmanageable. OsmAnd seems like a perfectly natural place to aggregate information like this, especially since they're already planning to add mixed-mode routing #7463 )
